nuxeo-core/nuxeo-core-bulk/src/main/java/org/nuxeo/ecm/core/bulk/action/computation/AbstractBulkComputation.java

```diff
@@ -85,24 +85,31 @@ public AbstractBulkComputation(String name, int nbOutputStreams) {
     @Override
     public void processRecord(ComputationContext context, String inputStreamName, Record record) {
         BulkBucket bucket = BulkCodecs.getBucketCodec().decode(record.getData());
-        command = getCommand(bucket.getCommandId());
-        if (command != null) {
-            startBucket(record.getKey());
-            for (List<String> batch : Lists.partition(bucket.getIds(), command.getBatchSize())) {
-                processBatchOfDocuments(batch);
-            }
-            endBucket(context, bucket.getIds().size());
-            context.askForCheckpoint();
-        } else {
-            if (isAbortedCommand(bucket.getCommandId())) {
-                log.debug("Skipping aborted command: {}", bucket.getCommandId());
+        LoginContext loginContext = null;
```

Comment:
Any way to avoid this extra sudo for each record ? 
This adds a new constraint to actor computation that are not AbstractBulkComputation, sounds like a pitfall.

Reply:
outdated